network/hive.go

```diff
@@ -83,7 +83,7 @@ func NewHive(params *HiveParams, kad *Kademlia, store state.Store) *Hive {
 // server is used to connect to a peer based on its NodeID or enode URL
 // these are called on the p2p.Server which runs on the node
 func (h *Hive) Start(server *p2p.Server) error {
-	log.Info("Starting hive", "baseaddr", fmt.Sprintf("%x", h.BaseAddr()[:4]))
+	log.Info("Starting hive", "baseaddr", fmt.Sprintf("%x", h.BaseAddr()[:4]), "discovery", h.Discovery)
```

Comment:
What you want is not start the connect loop on  100 below if `Discovery` is set to false

Reply:
I consider that, but the flag being used on line 133 confused me. I think the logic would still be the same if we don't start the `connect loop` as it is now, so the tests would still fail, so I have to investigate that first.